eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/UnitTests/IncrementalStakeTest.hs

```diff
@@ -0,0 +1,132 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Test.Cardano.Ledger.Shelley.UnitTests.IncrementalStakeTest where
+
+import Cardano.Ledger.Compactible (fromCompact)
+import Cardano.Ledger.Core (Era (..), EraTxOut, emptyPParams)
+import Cardano.Ledger.EpochBoundary (SnapShot (..), Stake (..))
+import Cardano.Ledger.Shelley (Shelley)
+import Cardano.Ledger.Shelley.LedgerState (
+  IncrementalStake (..),
+  dsUnifiedL,
+  incrementalStakeDistr,
+  updateStakeDistribution,
+ )
+import qualified Cardano.Ledger.UMap as UM
+import Data.Map.Strict (Map)
+import qualified Data.Map.Strict as Map
+import qualified Data.VMap as VMap
+import Lens.Micro
+import Test.Cardano.Ledger.Common
+import Test.Cardano.Ledger.UnitTestTools
+
+ppIdL :: Lens' (PoolParams c) (KeyHash 'StakePool c)
+ppIdL = lens ppId (\x y -> x {ppId = y})
+
+psStakePoolParamsL ::
+  Lens' (PState era) (Map (KeyHash 'StakePool (EraCrypto era)) (PoolParams (EraCrypto era)))
+psStakePoolParamsL = lens psStakePoolParams (\x y -> x {psStakePoolParams = y})
+
+{-
+InQuickCheck we might say
+property (assertBool (3==4)) :: Property
+in Hspec we might say
+prop "adc" (assertBool "3==4" (3==4)) :: Spec
+or we could say
+prop "abc" (True === False) :: Spec
+
+-}
+-- ===========================================
+-- A running example
+
+test1 :: forall era. EraTxOut era => Gen Property
+test1 = do
+  (tom, john, ann, ron, mary) <- fresh @5 @(Credential 'Staking (EraCrypto era))
+  (pool1, pool2) <- fresh @2 @(KeyHash 'StakePool (EraCrypto era))
+  pool1Params <- arbitraryLens ppIdL pool1
+  pool2Params <- arbitraryLens ppIdL pool2
+
+  (tomRD, johnRD, annRD, ronRD, maryRD) <- fresh @5 @RDPair
+
+  let tomAddr = addr tom Nothing -- Nothing means tomAddr does not have a StakeReference
+      johnAddr = addr john (Just john)
+      annAddr = addr ann (Just ann)
+      ronAddr = addr ron (Just ron)
+      _maryAddr = addr mary Nothing
```

Comment:
I am sure you are planning to remove those bindings with underscore, but just in case I'll post here a reminder.
```suggestion
```